packages/react-dialog/src/components/Dialog/useDialog.ts

```diff
@@ -12,17 +13,42 @@ import type { DialogProps, DialogState } from './Dialog.types';
  * @param ref - reference to root HTMLElement of Dialog
  */
 export const useDialog_unstable = (props: DialogProps, ref: React.Ref<HTMLElement>): DialogState => {
-  return {
-    // TODO add appropriate props/defaults
+  const { overlay, open = false, type = 'modal' } = props;
+  const isNonModal = type === 'non-modal';
+
+  const contentRef = React.useRef(null);
+  const { modalAttributes } = useModalAttributes({ trapFocus: !isNonModal && true });
+  const { findFirstFocusable } = useFocusFinders();
+
+  React.useEffect(() => {
+    if (open && contentRef.current) {
+      const firstFocusable = findFirstFocusable(contentRef.current);
+      firstFocusable?.focus();
+    }
+  }, [open, contentRef, findFirstFocusable]);
+
+  const state: DialogState = {
+    open,
+    type,
     components: {
-      // TODO add slot types here if needed (div is the default)
       root: 'div',
+      overlay: 'div',
     },
-    // TODO add appropriate slots, for example:
-    // mySlot: resolveShorthand(props.mySlot),
+
     root: getNativeElementProps('div', {
-      ref,
+      ref: useMergedRefs(ref, contentRef),
+      role: 'dialog',
+      'aria-modal': !isNonModal && true,
+      'aria-label': props['aria-labelledby'],
```

Comment:
`aria-label` and `aria-labelledby` are different concepts, looks like this should be removed and we should only rely on spreading of all props

Reply:
We should however make sure that `aria-label` or `aria-labelledby` is set per the WAI design pattern: https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html

I'm not entirely sure if this can be done with types (it would be great it possible) but perhaps a runtime warning (non-production) would work

```ts
if (!props['aria-label'] && !props['aria-labelledby'] && process.env.NODE_ENV !== 'production') {
  console.error('please apply aria-label or aria-labelledby to the Dialog')
}
```